Lib/http/cookiejar.py

```diff
@@ -532,15 +532,31 @@ def parse_ns_headers(ns_headers):
     return result
 
 
+# only kept for backwards compatibilty.
 IPV4_RE = re.compile(r"\.\d+$", re.ASCII)
+
+def is_ip(text: str):
+    """Return True if text is a valid IP address."""
+    from ipaddress import IPv4Address, IPv6Address
+    # check for IPv4 address
+    try:
+        IPv4Address(text)
+    except ValueError:
+        # check for IPv6 address in []
+        if text.startswith('[') and text.endswith(']'):
+            try:
+                IPv6Address(text.removeprefix('[').removesuffix(']'))
+            except ValueError:
+                return False
+        else:
+            return False # not a IPv6 address in []
```

Comment:
Why are IPv6 addressed not surrounded `[]` rejected? when making a request, we would use `http://[::1]` for instance, but we could decide to reject the domain `::1` itself. Please be more careful. Instead, we should make it allow both valid IPv6 addresses as well as IPv6 addresses enclosed in `'[]'`. But then, we need to be careful with the matching. Should `[::1]` be matched against `::1` or `[::1]` only? 
 
Try to look at the RFC as well to see if there are more information about IPv6 domains.